src/main/java/com/spread/progress/IProgressReporter.java

```diff
@@ -0,0 +1,6 @@
+package com.spread.progress;
+
+public interface IProgressReporter {
+   void registerProgressObserver(IProgressObserver observer);
+   void updateProgress(double progress);
```

Comment:
IMO updateProgress doesn't belong to the interface, since no one is going to call this from outside. updateProgress if we define it should be a private method of the classes that implement IProgressReporter, just as a helper method for calling the registered observer callback.